### PR TITLE
fix(sessiond): fix race condition on handle_activate_ue_flows_callback

### PR DESCRIPTION
**Pull request by uri200**
## Summary
backport of https://github.com/magma/magma/pull/10211

## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

**Review by themarwhal — APPROVED** [review]

**Review by ulaskozat — CHANGES_REQUESTED** [review]
needs verification on spirent

File: lte/gateway/c/session_manager/LocalEnforcer.cpp [hunk 1]
```diff
@@ -1469,8 +1470,7 @@ void LocalEnforcer::handle_set_session_rules(
 
       // Propagate these rule changes to PipelineD and MME (if 4G)
       propagate_rule_updates_to_pipelined(
-          session->get_session_id(), config, pending_activation,
-          pending_deactivation, false);
```
Comment (uri200) [inline lte/gateway/c/session_manager/LocalEnforcer.cpp]: it looks like i was passing a sting directly not a reference. Not sure why it was not crashing

**Comment by uri200** [comment]
> needs verification on spirent

verified around 10 times.  I saw same numbers on mme, a1ap, table 0 and sessiond. 

Added `sessiond_state` to `state_check.sh`

```
magma@u-160-161-nonnat:~$ sudo ./state_check.sh
MME_state = 206
S1AP_state = 206
SPGW_state = 206
Sessiond_state = 206
table0 DL bearers = 206
table12 DL flows = 206 ; table12 UL flows = 206
table13 DL flows = 412 ; table13 UL flows = 412
Mobilityd state = 210
```

**Review by ulaskozat — APPROVED** [review]
Thanks for the runs. I will merge after sanity confirmations.

**Pull request merged**